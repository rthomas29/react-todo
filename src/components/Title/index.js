import React from 'react'

// util function to render correct title based on num of tasks
const getHeaderTitle = (numTasks) => {
  if (numTasks > 0) {
    if (numTasks === 1) {
      return `${numTasks} task remaining`
    } else {
      return `${numTasks} tasks remaining`
    }
  } else {
    return 'No tasks remaining ğŸ¤—'
  }
}

const Title = ({ numTasks }) => {
  const headerTitle = getHeaderTitle(numTasks)
  return (
    <h2>{headerTitle}</h2>
  )
}

export default Title