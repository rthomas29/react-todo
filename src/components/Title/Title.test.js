import React from 'react'
import { render } from 'react-testing-library'
import Title from './index'

describe('<Title />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Title />)
    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should match snapshot accurate title if no tasks remain', () => {
    const { container } = render(<Title numTasks={0} />)
    expect(container.firstChild.innerHTML).toEqual('No tasks remaining ðŸ¤—')
  });
  it('should match snapshot accurate title if one task remains', () => {
    const { container } = render(<Title numTasks={1} />)
    expect(container.firstChild.innerHTML).toEqual('1 task remaining')
  });

  it('should match snapshot accurate title if more than one task remains', () => {
    const { container } = render(<Title numTasks={3} />)
    expect(container.firstChild.innerHTML).toEqual('3 tasks remaining')
  });
})