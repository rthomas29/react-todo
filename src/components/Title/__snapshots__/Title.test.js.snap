// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should match snapshot 1`] = `
<h2>
  No tasks remaining ðŸ¤—
</h2>
`;
